src/main/java/com/conveyal/r5/transit/fare/FareTable.java

```diff
@@ -47,7 +54,9 @@ public Fare lookup (String from, String to) {
 
     public Fare lookup (Stop from, Stop to) {
         //TODO: how does this works WRT different GTFS feeds having Stops with same Ids?
-        return lookup(from.stopId, to.stopId);
+        return this.ignoreAgencyId
+                ? lookup(from.stopId.split(":")[1], to.stopId.split(":")[1])
```

Comment:
Absolutely, we do need to avoid splitting on colons in the ID. But we don't need to manually construct a regex. The split function has a limit parameter to limit the number of splits. stopId.split(":", 2) should do it.